src/components/code/text_diff.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { HTMLAttributes, useMemo, StatelessComponent } from 'react';
+import Diff from 'text-diff';
+import classNames from 'classnames';
+import { CommonProps } from '../common';
+
+export interface EuiTextDiffSharedProps {
+  /**
+   * Sets the syntax highlighting for a specific language
+   * @see http://highlightjs.readthedocs.io/en/latest/css-classes-reference.html#language-names-and-aliases
+   * for options
+   */
+  language?: string;
+  overflowHeight?: number;
+  transparentBackground?: boolean;
+  isCopyable?: boolean;
+  /**
+   * passing a timeout of value '0' disables the timeout state
+   */
+  timeout?: number;
+}
+
+interface Props extends EuiTextDiffSharedProps {
+  currentText: string;
+  initialText: string;
+  InsertComponent?: StatelessComponent;
+  DeletionComponent?: StatelessComponent;
+  NoChangeComponent?: StatelessComponent;
+}
+
+export type EuiTextDiffProps = CommonProps &
+  Props &
+  HTMLAttributes<HTMLElement>;
+
+export const useEuiTextDiff: Function = ({
+  InsertComponent = 'ins',
+  DeletionComponent = 'del',
+  NoChangeComponent = 'span',
+  initialText = '',
+  currentText = '',
+  timeout = 0.1,
+  ...rest
+}) => {
```

Comment:
Since this is no longer a `FunctionComponent<props>`, TS doesn't know what these arguments are and needs to be typed:

```suggestion
}: EuiTextDiffProps) => {
```